### PR TITLE
use NewLedgerChangeReaderFromLedgerCloseMeta

### PR DESCRIPTION
**Pull request by sfsf9797**
would it be more efficient to use NewLedgerTransactionReaderFromLedgerCloseMeta?  feel free to close this if the PR doesn't make sense. Thanks.

### CODE REVIEW

File: internal/input/transactions.go [hunk 1]
```diff
@@ -33,17 +33,18 @@ func GetTransactions(start, end uint32, limit int64, env utils.EnvironmentDetail
 	err = backend.PrepareRange(ctx, ledgerbackend.BoundedRange(start, end))
 	panicIf(err)
 	for seq := start; seq <= end; seq++ {
-		txReader, err := ingest.NewLedgerTransactionReader(ctx, backend, env.NetworkPassphrase, seq)
```
Comment (chowbao) [inline internal/input/transactions.go:36]: Hmm actually the returned LedgerTransactionReader (the txReader in this case) has ledgerCloseMeta inside it. So the refactor should be to 
* remove the extra ledgerCloseMeta in line 43
* update LedgerCloseMeta in line 57 to be set by `txReader.ledgerCloseMeta`

Comment (sfsf9797) [reply internal/input/transactions.go:36]: seem like txReader.ledgerCloseMeta field is not exported?

 https://github.com/stellar/go/blob/master/ingest/ledger_transaction_reader.go#L17

Comment (chowbao) [reply internal/input/transactions.go:36]: I believe it should be
`NewLedgerTransactionReader()` calls `NewLedgerTransactionReaderFromLedgerCloseMeta()` which then sets  `ledgerCloseMeta` [here](https://github.com/stellar/go/blob/master/ingest/ledger_transaction_reader.go#L36)

Comment (sfsf9797) [reply internal/input/transactions.go:36]: oh sorry, I might have misunderstood it. 
what I am trying to say is the ledgerCloseMeta field is not exported as it is not in capital letters. 


image

Comment (chowbao) [reply internal/input/transactions.go:36]: Ah I missed that. You're right then

**Review by chowbao — APPROVED** [review]

**Pull request merged**